src/main/java/com/puppycrawl/tools/checkstyle/api/BeforeExecutionFileFilter.java

```diff
@@ -0,0 +1,33 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2016 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.api;
+
+/**
+ * An interface for before execution file filtering events.
+ * @author Richard Veach
+ */
+public interface BeforeExecutionFileFilter {
+    /**
+     * Determines whether or not a before execution file filtered event is accepted.
+     * @param event the BeforeExecutionFileFilterEvent to filter.
+     * @return true if the event is accepted.
+     */
+    boolean accept(BeforeExecutionFileFilterEvent event);
```

Comment:
Do we really need BeforeExecutionFileFilterEvent ?

Why not simple `boolean accept(String uri);` ? What else xxxxxxxxxxxFileFilter could care about ?
https://prateekvjoshi.com/2014/02/22/url-vs-uri-vs-urn/ . We just need string representation of location.